src/components/data/CreateFolderDialog.js

```diff
@@ -0,0 +1,165 @@
+                                <Close />
+                            </IconButton>
+                        </DialogTitle>
+                        <DialogContent>
+                            <Field
+                                id={build(baseId, ids.FOLDER_NAME)}
+                                name="name"
+                                multiline
+                                required={true}
+                                label={getMessage("folderName")}
+                                onKeyDown={(event) => {
+                                    if (event.key === "Enter") {
+                                        handleSubmit();
+                                    }
+                                }}
+                                InputProps={{
+                                    readOnly: isLoading,
+                                    endAdornment: (
+                                        <>
+                                            {isLoading && (
+                                                <InputAdornment position="start">
+                                                    <CircularProgress
+                                                        id={build(
+                                                            baseId,
+                                                            ids.FOLDER_NAME,
+                                                            ids.LOADING_SKELETON
+                                                        )}
+                                                        color="inherit"
+                                                        size={20}
+                                                    />
+                                                </InputAdornment>
+                                            )}
+                                        </>
+                                    ),
+                                }}
+                                component={FormTextField}
+                            />
+                        </DialogContent>
+
+                        <DialogActions>
+                            <Button
+                                id={build(baseId, ids.CANCEL_BTN)}
+                                onClick={onClose}
+                            >
+                                {getMessage("cancel")}
+                            </Button>
+                            <Button
+                                id={build(baseId, ids.CREATE_BTN)}
+                                color="primary"
+                                type="submit"
+                                onClick={handleSubmit}
+                            >
+                                {getMessage("create")}
+                            </Button>
+                        </DialogActions>
+                    </Dialog>
+                </Form>
+            )}
+        </Formik>
+    );
+}
+
+export default withI18N(injectIntl(CreateFolderDialog), messages);
```

Comment:
I think we should add the `parent folder` path to this dialog to inform the user where the folder is being created.

Reply:
One other concern with Dialogs in general, if there is a service error (in this case, the folder already exists), we are displaying an error announcement. But the modal dialog makes it hard to realize that there was an announcement in the background (i tried this PR). Should we display these errors in the dialog itself just like how validation messages are displayed? Some thing to think about. Not a deal breaker for this PR.

Reply:
Good point. I think I can refactor so that it will check for existing folder names and display an error within the dialog.

I also realized we should probably be checking permissions before displaying this button, so I'll add that as well.